### DISCUSSION TITLE
htmlは毎回出力すべきか

### DISCUSSION THREAD

**Discussion by sakots** [satopian/poti-kaini#8] [Ideas]
POTIが出た2005年頃は、CPUがPentium4とかPentiumDの時代です。

現在のCPUはその2倍くらいの周波数で、しかも8スレッドとか16スレッド、サーバー用マシンでは32スレッドくらいの命令が動いています。

今なら100回potiboard.phpを読み込んで命令を動かすよりも、100枚のhtmlを書き換える負荷のほうが高いのではないかと思っています。

なので、html書き出し機能は過去ログ(アーカイブ)生成のときだけ、とも思ったんですがどうでしょうか。

**Comment by satopian** [comment]
いまから、HTML出力はしませんよとした時。
POTIでログが飛んだ時の過去の対処方法をみると、HTMLファイルは残っていたからそこから拾うといった事もあります。
PHPのバージョンがあがって動かなくなった掲示板が多いなか、台湾のPOTI-boardはHTMLだけは残っていました。
poo掲示板を借りていたひとは、手作業で毎回HTML化していました。
そう考えると、常に過去ログの準備をしているようなところもあるので、PHP表示がいいと割り切れません。
8000発言ぐらいで掲示板を入れ替えていますが、前の掲示板は何もしなくても、HTML化されているので実質そのまま過去ログです。
そして、POTI-boardの過去ログ機能が開発できていないので、そこだけHTML化すればいいんじゃないの?も実現していません。
現時点ではHTML出力のまま行く予定です。
それから、これはftpでこまめにHTMLファイルを削除できる人はいいのですが、すでにHTML化されたファイルの更新を停止してphp出力にすると、発言を削除してもHTMLファイルは更新されずに残るしクローラーも読み取る。
古い発言が残っているつもりじゃなかったとしても、いくつものHTMLファイルが残る可能性があります。
これはHTMLで出力しますというPOTIの最初の仕様の問題なので、新しく作る掲示板ではそのあたりの問題はないのかもしれません。
KENT-WEBさんの掲示板は過去ログだけHTMLですが、だったらそうすればも、Tweetのリンク切れ問題も発生するので、うまくまとまっていません。
そして、そこにあまり時間をさいて何かをやりたいという現状でもない感じです。
過去ログの仕様もあとからやっぱりはさけたいですし、時間をかけて考えてからですね。

**Comment by sakots** [comment]
なるほど。難しいですね…

**Comment by satopian** [comment]
ある程度使い物になる完成度にはなっているので、今は全世界にオエビを広めるべく宣伝に力をそそぎたい感じですね。
これは困るという点があればユーザーから何とかして欲しいという要望が出てきてますし、そこはなんとか対応している感じです。
過去ログ機能があったほうがいいのか、余計な事はしないほうがいいのかは結論がでていないのですが、数ヶ月で考えがまとまる気がしないので、いろいろと学習する過程である日突然みたいな感じになるのかもしれません。

**Comment by sakots** [comment]
それはわかっていますので、issueにはしませんでした。

**Comment by sakots** [comment]
たしかにまあログが1000ページ（1ページ10スレッドなら絵が10000枚）の掲示板がある方がおかしいっちゃおかしいんですけども

**Comment by satopian** [comment]
textのimg.logの上限を5MBに設定しているので、それ以上はログ保持件数をいくら増やしても保存されないですね。
以前は上限が1MBだったので、4000発言でログが消えるのはなぜ?という質問もでていました。(ぷにゅねっと
どうやったらいいのかわかりませんが、古い発言の分までHTMLを更新しなくてはいけないという仕様なのを、追記してファイルが増えていくだけにできれば…。
そういう意味では古い発言はまとめて別フォルダへHTMLを移動して過去ログ化(更新しないHTML)にすれば新規投稿時の負荷も下がると思います。
ただ、それをやるだけの技術は今の所ありません。
いろいろなスクリプトを読んで、発想のヒントが何かないか調べているところです。

**Comment by sakots** [comment]
昔きつねこさんが言ってた（きがする）「htmlファイルの名前だけ計算して予約しておいて、リクエストがあってから生成する」ですかねえ…
skinnyの命令で一気に吐き出してるんで、ココをforeachとかにするのかなあ

**Comment by satopian** [comment]
> skinnyの命令で一気に吐き出してるんで、ココをforeachとかに

すでにそういう構造です。
for文で出力するページ分ループして1ページづつHTML出力を繰り返す。
skinnyでやっているのは、1ページ分のHTMLをちょうだいでしかありません。
コードに関する理解の違いがあると議論がすすまないので説明しなければならなくなってしまいますが、その時間はちょっと無いのでこの話はこの辺にしたいと思っています。
すみません。
